edgetest/utils.py

```diff
@@ -215,6 +215,23 @@ def parse_cfg(filename: str = "setup.cfg", requirements: Optional[str] = None) -
         if section_name[1] == "envs":
             output["envs"].append(dict(config[section]))
             output["envs"][-1]["name"] = section_name[2]
+            if (
+                "lower" in output["envs"][-1]
+                and "options" in config
+                and "install_requires" in config["options"]
+            ):
+                output["envs"][-1]["lower"] = "\n".join(
+                    f"{pkg_name}=={lower_bound}"
+                    for pkg_name, lower_bound in get_lower(
+                        config.get("options", "install_requires")
+                    ).items()
+                    if _isin_case_dashhyphen_ins(
+                        pkg_name, output["envs"][-1]["lower"].split("\n")
+                    )
+                    and lower_bound is not None
+                    # TODO: Emit warning if lower_bound is None but package is in lower
+                    # TODO: Parse through extra requirements as well to get lower bounds
+                )
```

Comment:
I'm finding this loop a bit difficult to read. What if we split it out a bit? Something along the lines of

```suggestion
                output["envs"][-1]["lower"] = ""
                for pkg_name, lower_bound in get_lower(config.get("options", "install_requires").items():
                    if (
                        _isin_case_dashhyphen_ins(pkg_name, output["envs"][-1]["lower"].split("\n")) and lower_bound is not None
                    ):
                        output["envs"][-1]["lower"] += f"{pkg_name}=={lower_bound}\n"
                    # TODO: Emit warning if lower_bound is None but package is in lower
                    # TODO: Parse through extra requirements as well to get lower bounds
```